### PR TITLE
Update API docs to reflect Grants API cleanup

### PR DESCRIPTION
**Pull request by bellatumaneng-okta**
## Description:
- Accompanying changes for https://github.com/okta/okta-core/pull/28254 to clean up Grants API
- Add correct `issuer` value and remove `issuerId`
- Remove `"issuer"` and `"revoke"` HAL links and add hints for `"self"` for `GET` and `DELETE`

### Resolves:
* [OKTA-175296](https://oktainc.atlassian.net/browse/OKTA-175296)

### CODE REVIEW

**Comment by jakubvul** [comment]
@bellatumaneng-okta which release is this supposed to go out with?

**Comment by bellatumaneng-okta** [comment]
The code changes haven't merged yet because I'm waiting for feedback from Product. It'll probably be 2018.30 if I don't get approval from that side today.

File: _source/_docs/api/resources/users.md [hunk 1]
```diff
@@ -4249,7 +4225,7 @@ For an individual User result, the Links Object contains a full set of link rela
 | status      | Status of the grant. Valid values: `ACTIVE`, `REVOKED` or `EXPIRED`                                                            | String                                                          |
 | created     | Timestamp when the grant was created                                                                                           | Date                                                            |
 | lastUpdated | Timestamp when the grant was last updated                                                                                      | Date                                                            |
-| issuerId    | ID of the authorization server for this grant                                                                                  | String                                                          |
+| issuer      | The complete URL of the authorization server for this grant                                                                    | String                                                          |
```
Comment (danielzeller-okta) [inline _source/_docs/api/resources/users.md]: Is this how we describe the issuer in other places in the docs? Seems a little clunky

File: _source/_docs/api/resources/users.md [hunk 2]
```diff
@@ -4184,52 +4166,46 @@ For an individual User result, the Links Object contains a full set of link rela
 
 ~~~sh
 {
-    “id”: “oag2n8HU1vTmvCdQ50g3",
```
Comment (danielzeller-okta) [inline _source/_docs/api/resources/users.md]: any clue why we were using these different quotation mark characters? +1 to cleanup

Comment (jakubvul) [reply _source/_docs/api/resources/users.md]: Yea we should be using `"` only. None of these fancy `“` and `”`. 😄

**Comment by bellatumaneng-okta** [comment]
FYI @jakubvul these changes should be going with this week's release. Thanks!

File: _source/_docs/api/resources/users.md (hunk 1 above)
Comment (jakubvul) [reply _source/_docs/api/resources/users.md]: We don't have a consistent way of describing this property: https://developer.okta.com/docs/api/resources/oidc

**Comment by jakubvul** [comment]
@bellatumaneng-okta I don't see it in the list for this week's release: https://oktawiki.atlassian.net/wiki/spaces/eng/pages/538020070/Release+Notes+2018.29

Also, you will need to clean up the non-unicode characters before you merge this.

Attn @eklein-okta

File: _source/_docs/api/resources/users.md (hunk 1 above)
Comment (bellatumaneng-okta) [reply _source/_docs/api/resources/users.md]: I used the phrasing from here: https://developer.okta.com/docs/api/resources/oidc#response-properties-6

**Comment by bellatumaneng-okta** [comment]
@jakubvul sorry, I was referring to this week's code freeze, so 2018.30.

I _think_ I cleaned up all the funny quotes in every snippet I modified for the current changes, but if you saw something else let me know and I can double check.

**Comment by jakubvul** [comment]
Indeed! Sorry, I misread the diff.

**Comment by nbarbettini** [comment]
Don't merge: waiting for 2018.31

**Pull request merged**